src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -956,6 +981,14 @@ private static JsonSerializerOptions GetOrCreateDefaultOptionsInstance()
             return Interlocked.CompareExchange(ref s_defaultOptions, options, null) ?? options;
         }
 
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
+        private static JsonSerializerOptions GetOrCreateWebOptionsInstance()
+        {
+            var options = new JsonSerializerOptions(JsonSerializerDefaults.Web) { _isReadOnly = true };
```

Comment:
You also need to set the `TypeInfoResolver` like the above method, otherwise this won't work as expected.